### PR TITLE
CF-426: add /status URI, without any roles required, for monitoring

### PR DESCRIPTION
**Pull request by shintasmith**
Add /status, for @mwhahaha :-)

### CODE REVIEW

**Comment by shintasmith** [comment]
Well, dang, the test broke on Jenkins. It ran fine on my box. So.... gotta debug this.

File: app/src/test/scala/com/rackspace/prefs/StatusTest.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        clearData(db)
+    }
+
+    test("should get 200: GET /status") {
+        addServlet(new PreferencesService(db), "/*")
+        get("/status") {
+            status should equal (200)
+            val prefStatus = JSON.parseFull(body)
+            val statusMap = prefStatus.get.asInstanceOf[Map[String, Int]]
+            statusMap.get("metadata-count").get should equal (1.0)
```
Comment (ChandraAddala) [inline app/src/test/scala/com/rackspace/prefs/StatusTest.scala]: I know we dont have plans to add another metatadata but something tells me, the condition could be "> 0" to be safe.

**Comment by shintasmith** [comment]
@ChandraAddala, good point, I updated the test to check for > 0. 

I also updated the one negative test so it puts back the metadata it deletes.

**Comment by usnavi** [comment]
lgtm; @ChandraAddala can merge when ready.

File: app/src/test/scala/com/rackspace/prefs/StatusTest.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    override def beforeAll {
+        super.beforeAll
+        clearData(db)
+    }
+
+    test("should get 200: GET /status") {
+        addServlet(new PreferencesService(db), "/*")
+        get("/status") {
+            status should equal (200)
+            val countConversionFunc = {input: String => new Integer(input)}
```
Comment (ChandraAddala) [inline app/src/test/scala/com/rackspace/prefs/StatusTest.scala]: You can do something like this to convert to integer. "5".toInt

File: app/src/test/scala/com/rackspace/prefs/StatusTest.scala [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            JSON.perThreadNumberParser = countConversionFunc
+            val prefStatus = JSON.parseFull(body)
+            val statusMap = prefStatus.get.asInstanceOf[Map[String, Int]]
+            statusMap.get("metadata-count").get should be > (0)
+            println(body)
```
Comment (ChandraAddala) [inline app/src/test/scala/com/rackspace/prefs/StatusTest.scala]: ha ha...you forgot to remote the println

File: app/src/test/scala/com/rackspace/prefs/StatusTest.scala [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    // this needs to be the last test run, cuz it's going to drop
+    // the Preferences Metadata table
+    test("should get 500: GET /status when metadata table does not exist") {
+        dropMetadata(db)
+        try {
+            get("/status") {
+                status should equal(500)
+                println(body)
```
Comment (ChandraAddala) [inline app/src/test/scala/com/rackspace/prefs/StatusTest.scala]: same here.

File: app/src/test/scala/com/rackspace/prefs/StatusTest.scala (hunk 3 above)
Comment (shintasmith) [inline app/src/test/scala/com/rackspace/prefs/StatusTest.scala]: Eh, it's a test code. :-) But I'll remove it.

File: app/src/test/scala/com/rackspace/prefs/StatusTest.scala (hunk 2 above)
Comment (shintasmith) [inline app/src/test/scala/com/rackspace/prefs/StatusTest.scala]: Nope, I got ClassCast exception from Double to Int. The default numeric conversion function of the JSON parser is Double. Once you got Double back, it's hard to convert to Int. The Scala test "should be" doesn't work well with Double. This is what they recommend you do, if you want to get back Integer, from parsing JSON numeric.

**Comment by ChandraAddala** [comment]
lgtm;

**Pull request merged**